File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/ForkBasic.pl
#!/usr/bin/perl -w

### ForkBasic.pl

use strict;

my $child_pid = fork();                                            #(A)

die "Unable to fork: $!" unless defined $child_pid;                #(B)

my $str = "hello";                                                 #(C)

if ($child_pid == 0) {                                             #(D)
    # Execute in child process:
    print "I am the child process of PID $$\n\n";                  #(E)
    $str = "jello";                                                #(F)
} else {                                                           #(G)
    # Execute in parent process:
    sleep(1);                                                      #(H)
    print "      I am the parent process\n".                       #(I)
          "      My own PID is $$\n".                            
          "      My child's PID is $child_pid\n";

    print "      $str\n";              # hello                     #(J)
}

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/ForkBasic.py
#!/usr/bin/python

### ForkBasic.py

import os, time                                                     #(A)

pid = os.getpid()                                                   #(B)

message = 'hello'                                                   #(C)

child_pid = os.fork()                                               #(D)

if (child_pid == 0):                                                #(E)
    # Execute in child process:
    print 'I am a child of process %s\n' % pid                      #(F)
    message = "jello";                                              #(G)
else:                                                               #(H)
    # Execute in parent process:
    time.sleep(1);                                                  #(I)
    print "      I am the parent process\n" +         \
          "      My own PID is %s \n" % pid +         \
          "      My child's PID is %s\n" % child_pid                #(J)

    print "      %s\n" % message              # hello               #(K)
